src/app/georeferencer/qgsgeorefdatapoint.cpp

```diff
@@ -92,13 +92,18 @@ void QgsGeorefDataPoint::setEnabled( bool enabled )
 
 void QgsGeorefDataPoint::setId( int id )
 {
+  const bool noLongerTemporary = mId < 0 && id >= 0;
```

Comment:
Variable name doesn't seem optimal, maybe "isValid" or "isPermanent"?

Reply:
We only want to change the color if this call changes the point form a temporary ( < 0 ) to a permanent, hence the _noLonger_ in the name